pkg/segment/query/iqr/intermediateQueryResult.go

```diff
@@ -534,10 +537,8 @@ func MergeIQRs(iqrs []*IQR, less func(*Record, *Record) bool) (*IQR, int, error)
 			iqr.rrcs = append(iqr.rrcs, record.iqr.rrcs[record.index])
 		}
 		for cname, values := range record.iqr.knownValues {
-			if _, ok := iqr.knownValues[cname]; !ok {
-				err := toputils.TeeErrorf("MergeIQRs: column %v is missing from destination IQR", cname)
-				return nil, 0, err
-			}
+			// we should not validate cname in the destination iqr because caching
+			// will have populated the knownValue in the source IQR
 			iqr.knownValues[cname] = append(iqr.knownValues[cname], values[record.index])
```

Comment:
I don't understand this; can you explain it to me?

Reply:
Say for example when we merged the metaData, the IQRs did not have any known values since they were withRRCs IQRs, what happened was that IndexOfMin got called which read the column value to sort `first_name` in the given query and cached it in the knownValues. Now, source IQR got populated with knownValue but destination IQR did not. This gave the error and query was not executing correctly.

Reply:
Ah ok. I was thinking that in the case where we would return an error now we would panic, but apparently that doesn't happen: https://go.dev/play/p/xm_BPHWwh4y